.github/workflows/generate-tidal-api.yml

```diff
@@ -0,0 +1,122 @@
+name: Generate API in the TidalAPI module and create a PR if there are changes
+
+on:
+  workflow_dispatch: # manual triggering of this workflow
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  codegen:
+    runs-on: macos-latest
+    env:
+      API_FOLDER: Sources/TidalAPI/
+      GENERATED_FOLDER_NAME: Generated
+      INPUT_FOLDER_NAME: Config/input
+      SCRIPT_NAME: generate_and_clean.sh
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      # Step 2: Install OpenAPI Generator via Homebrew
+      - name: Install OpenAPI Generator with Homebrew
+        run: |
+          brew install openapi-generator
+
+      - name: Run Code Generation Script
+        run: |
+          cd $API_FOLDER
+          chmod +x $SCRIPT_NAME
+          ./$SCRIPT_NAME
+
+      - name: Check for changes
+        id: check_for_changes
+        run: |
+          cd $API_FOLDER
+          git add . 
+          changes_in_generated=$(git status -s $GENERATED_FOLDER_NAME | sed 's/"//g')
+          echo "Changes in $GENERATED_FOLDER_NAME:"
+          echo "$changes_in_generated"
+
+          changes_in_input=$(git status -s $INPUT_FOLDER_NAME | sed 's/"//g')
+          echo "Changes in $INPUT_FOLDER_NAME:"
+          echo "$changes_in_input"
+
+          if [ -n "$changes_in_generated" ] || [ -n "$changes_in_input" ]; then
+            changes_detected="true"
+          else
+            changes_detected="false"
+          fi
+          echo "Changes detected: $changes_detected"
+
+          {
+            echo 'CHANGES_IN_GENERATED<<EOF'
+            echo "$changes_in_generated"
+            echo 'EOF'
+          } >> $GITHUB_OUTPUT
+
+          {
+            echo 'CHANGES_IN_INPUT<<EOF'
+            echo "$changes_in_input"
+            echo 'EOF'
+          } >> $GITHUB_OUTPUT
+
+          echo "CHANGES_DETECTED=$changes_detected" >> $GITHUB_OUTPUT
+
+      - name: No Changes Found - Say Goodbye
+        if: ${{ steps.check_for_changes.outputs.CHANGES_DETECTED == 'false' }}
+        run: |
+            echo "No changes found. Good bye"
```

Comment:
Instead of this, and all the extra `if:` checks on every step, you could do this in the previous step (pseudo-code)

```
if [ "$changes_detected" = "false" ]; then
  echo "No changes found. Good bye"
  exit 0
fi
```

Reply:
It doesn't work, further steps are still tried to execute and fail.

(During my development `exit 1` worked though, but the workflow was also failing, and I'd like to avoid it)

Maybe I am doing something wrong? 🤔  

https://github.com/tidal-music/tidal-sdk-ios/actions/runs/11572480528/job/32212562473
![Screenshot 2024-10-29 at 12 03 30](https://github.com/user-attachments/assets/26c7df3c-837e-4e7a-b719-20a0802d478f)

Reply:
No, seems like you are correct
https://github.com/actions/runner/issues/662

I can't believe they don't have a way of doing this.

Reply:
🤷‍♂️

Reply:
Agree, it is such an obvious functionality to have